src/Game/zCamera.cpp

```diff
@@ -2,37 +2,179 @@
 
 #include <types.h>
 
+#include "zCameraTweak.h"
+#include "../Core/x/xVec3Inlines.h"
+
+enum WallJumpViewState
+{
+	WJVS_DISABLED,
+	WJVS_DISABLING,
+	WJVS_ENABLED,
+	WJVS_ENABLING
+};
+
+struct zFlyKey
+{
+	int32 frame;
+	float32 matrix[12];
+	float32 aperture[2];
+	float32 focal;
+};
+
+// extern int8 buffer[16];
+// extern int8 buffer[16];
+// extern basic_rect screen_bounds;
+// extern basic_rect default_adjust;
+extern float32 zcam_pad_pyaw_scale;
+extern float32 zcam_pad_pitch_scale;
+extern float32 zcam_near_d;
+extern float32 zcam_near_h;
+extern float32 zcam_near_pitch;
+extern float32 zcam_far_d;
+extern float32 zcam_far_h;
+extern float32 zcam_far_pitch;
+extern float32 zcam_wall_d;
+extern float32 zcam_wall_h;
+extern float32 zcam_wall_pitch;
+extern float32 zcam_above_d;
+extern float32 zcam_above_h;
+extern float32 zcam_above_pitch;
+extern float32 zcam_below_d;
+extern float32 zcam_below_h;
+extern float32 zcam_below_pitch;
+extern float32 zcam_highbounce_d;
+extern float32 zcam_highbounce_h;
+extern float32 zcam_highbounce_pitch;
+extern float32 zcam_overrot_min;
+extern float32 zcam_overrot_mid;
+extern float32 zcam_overrot_max;
+extern float32 zcam_overrot_rate;
+extern float32 zcam_overrot_tstart;
+extern float32 zcam_overrot_tend;
+extern float32 zcam_overrot_velmin;
+extern float32 zcam_overrot_velmax;
+extern float32 zcam_overrot_tmanual;
+extern float32 zcam_overrot_tmr;
+extern xCamera zcam_backupcam;
+extern xCamera zcam_backupconvers;
+extern int32 zcam_near;
+extern int32 zcam_mode;
+extern int32 zcam_bbounce;
+extern int32 zcam_lbbounce;
+extern int32 zcam_convers;
+extern int32 zcam_lconvers;
+extern int32 zcam_longbounce;
+extern int32 zcam_highbounce;
+extern int32 zcam_cutscene;
+extern int32 zcam_reward;
+extern xVec3* zcam_playervel;
+extern float32 zcam_mintgtheight;
+extern int32 zcam_fly;
+extern int32 zcam_flypaused;
+extern void* zcam_flydata;
+extern uint32 zcam_flysize;
+extern float32 zcam_flytime;
+extern uint32 zcam_flyasset_current;
+extern xCamAsset* zcam_dest;
+extern xQuat zcam_quat;
+extern float32 zcam_tmr;
+extern float32 zcam_ttm;
+extern float32 zcam_fovcurr;
+extern float32 zcam_fovdest;
+extern uint32 stop_track;
+extern uint8 input_enabled;
+extern uint8 lassocam_enabled;
+extern float32 lassocam_factor;
+extern WallJumpViewState wall_jump_enabled;
+extern xVec3 wall_jump_view;
+extern float32 dMultiplier;
+extern float32 dOffset;
+extern float32 hMultiplier;
+extern float32 hOffset;
+extern float32 rewardMove;
+extern float32 rewardMoveSpeed;
+extern float32 rewardZoomSpeed;
+extern float32 rewardZoomAmount;
+extern float32 rewardTiltTime;
+extern float32 rewardTiltAmount;
+// extern zGlobals globals;
+extern xVec3 g_O3;
+extern float32 gSkipTimeFlythrough;
+
+extern float32 lbl_803CD1A8;
+extern float32 lbl_803CD1AC;
+extern float32 lbl_803CD1B0;
+extern float32 lbl_803CD1B4;
```

Comment:
these however are constants. But i have no idea what their values are. Is there a way to find out other than estimating?

Reply:
Go to where they are defined in the assembly. It will have an incbin of 4 bytes. Go to the location in the baserom where it is importing the 4 bytes those 4 bytes are the float value, use a tool in a hex editor or online to get the value of those hex bytes to a 32 bit float

Reply:
If you want to rename them now, you can, and I'll wait to merge the PR. If you don't want to do it, let me know and I'll merge it now

Reply:
No please hold on with the merge. Ill do it and i also will include the non matching functions